File: LeetCode/Medium/1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int ret = 0;
        int prev = 0;
        int temp = 0;
        char[] sCharArray = s.toCharArray();
        
        for (int i = 0; i < k; ++i){
            if (sCharArray[i] == 'a' || sCharArray[i] == 'e' ||
               sCharArray[i] == 'o' || sCharArray[i] == 'i' || sCharArray[i] == 'u') ++temp;
        }
        
        ret = temp;
        if (sCharArray[prev] == 'a' || sCharArray[prev] == 'e' ||
               sCharArray[prev] == 'o' || sCharArray[prev] == 'i' || sCharArray[prev] == 'u') temp--;
        ++prev;
        for (int i = k; i < sCharArray.length; ++i){
            if (sCharArray[i] == 'a' || sCharArray[i] == 'e' ||
               sCharArray[i] == 'o' || sCharArray[i] == 'i' || sCharArray[i] == 'u') ++temp;
            
            ret = Math.max(ret, temp);
            
            if (sCharArray[prev] == 'a' || sCharArray[prev] == 'e' ||
               sCharArray[prev] == 'o' || sCharArray[prev] == 'i' || sCharArray[prev] == 'u') temp--;
            ++prev;
        }
        
        return ret;
    }
}

File: LeetCode/Medium/1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowel = set(["a","e","i","o","u"])
        ret = 0
        prev = 0
        temp = 0
        for i in range(k):
            if s[i] in vowel:
                temp += 1

        ret = temp

        if s[prev] in vowel:
            temp -= 1
        prev += 1
        
        for i in range(k, len(s)):
            if s[i] in vowel:
                temp += 1
            ret = max(ret, temp)
            if s[prev] in vowel:
                temp -= 1
            prev += 1
        return ret